apiserver/facades/agent/provisioner/provisioner_test.go

```diff
@@ -113,12 +114,13 @@ func (s *provisionerSuite) setUpTest(c *gc.C, withController bool) {
 	s.serviceFactory = s.ControllerServiceFactory(c)
 	// Create a provisioner API for the machine.
 	provisionerAPI, err := provisioner.NewProvisionerAPIV11(context.Background(), facadetest.ModelContext{
-		Auth_:           s.authorizer,
-		State_:          st,
-		StatePool_:      s.StatePool(),
-		Resources_:      s.resources,
-		ServiceFactory_: s.serviceFactory,
-		Logger_:         loggertesting.WrapCheckLog(c),
+		Auth_:            s.authorizer,
+		State_:           st,
+		StatePool_:       s.StatePool(),
+		Resources_:       s.resources,
+		ServiceFactory_:  s.serviceFactory,
+		Logger_:          loggertesting.WrapCheckLog(c),
+		WatcherRegistry_: facademocks.NewMockWatcherRegistry(gomock.NewController(c)),
```

Comment:
You can't do this in `SetUpTest`. It has a different `gc.C` than the test methods, which causes issues.

You are also not calling `Finish` on it.